'Constant name doesn't conform' warning on `global` statement instead of declaration

### Bug description
```python
import typing as _tp

_foo: _tp.Optional["Foo"] = None


class Foo:
    ...


def setup_shared_foo():
    global _foo

    _foo = Foo()
```

### Configuration

_No response_

### Command used

```shell
pylint test.py
```


### Pylint output

```shell
test.py:11:4: C0103: Constant name "_foo" doesn't conform to UPPER_CASE naming style ('([^\\W\\da-z][^\\Wa-z]*|__.*__)$' pattern) (invalid-name)
```


### Expected behavior

Pylint should warn about the 'incorrect' variable name on line 3, where the variable is defined, not line 11, where it is used in a `global` statement.

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]
```


### OS / Environment

Windows 10

### Additional dependencies

_No response_

--- a/doc/whatsnew/fragments/8307.bugfix
+++ b/doc/whatsnew/fragments/8307.bugfix
@@ -1,0 +1,3 @@
+Prevent emitting ``invalid-name`` for the line on which a ``global`` statement is declared.
+
+Closes #8307

--- a/pylint/checkers/base/name_checker/checker.py
+++ b/pylint/checkers/base/name_checker/checker.py
@@ -388,11 +388,6 @@ class NameChecker(_BasicChecker):
 
     visit_asyncfunctiondef = visit_functiondef
 
-    @utils.only_required_for_messages("disallowed-name", "invalid-name")
-    def visit_global(self, node: nodes.Global) -> None:
-        for name in node.names:
-            self._check_name("const", name, node)
-
     @utils.only_required_for_messages(
         "disallowed-name",
         "invalid-name",

--- a/tests/functional/g/globals.py
+++ b/tests/functional/g/globals.py
@@ -102,3 +102,12 @@ def init_connection_state(alias):
     global RAN_DB_DICT  # [global-variable-not-assigned]
     RAN_DB_SET.add(alias)
     return RAN_DB_DICT.setdefault("color", "Palomino")
+
+
+# Prevent emitting `invalid-name` for the line on which `global` is declared
+# https://github.com/PyCQA/pylint/issues/8307
+
+_foo: str = "tomato"
+def setup_shared_foo():
+    global _foo  # [global-statement]
+    _foo = "potato"

--- a/tests/functional/g/globals.txt
+++ b/tests/functional/g/globals.txt
@@ -14,3 +14,4 @@ global-statement:81:4:81:15:override_func:Using the global statement:HIGH
 global-statement:91:4:91:16:override_class:Using the global statement:HIGH
 global-variable-not-assigned:101:4:101:21:init_connection_state:Using global for 'RAN_DB_SET' but no assignment is done:HIGH
 global-variable-not-assigned:102:4:102:22:init_connection_state:Using global for 'RAN_DB_DICT' but no assignment is done:HIGH
+global-statement:112:4:112:15:setup_shared_foo:Using the global statement:HIGH

--- a/tests/functional/n/name/name_styles.py
+++ b/tests/functional/n/name/name_styles.py
@@ -94,7 +94,7 @@ NOT_CORRECT = CorrectClassName  # [invalid-name]
 def test_globals():
     """Names in global statements are also checked."""
     global NOT_CORRECT
-    global AlsoCorrect  # [invalid-name]
+    global AlsoCorrect
     NOT_CORRECT = 1
     AlsoCorrect = 2
 

--- a/tests/functional/n/name/name_styles.txt
+++ b/tests/functional/n/name/name_styles.txt
@@ -10,7 +10,6 @@ invalid-name:53:4:53:38:CorrectClassName.__DunDER_IS_not_free_for_all__:"Method 
 invalid-name:83:0:83:18::"Class name ""BAD_NAME_FOR_CLASS"" doesn't conform to PascalCase naming style":HIGH
 invalid-name:84:0:84:23::"Class name ""NEXT_BAD_NAME_FOR_CLASS"" doesn't conform to PascalCase naming style":HIGH
 invalid-name:91:0:91:11::"Class name ""NOT_CORRECT"" doesn't conform to PascalCase naming style":HIGH
-invalid-name:97:4:97:22:test_globals:"Constant name ""AlsoCorrect"" doesn't conform to UPPER_CASE naming style":HIGH
 invalid-name:110:4:110:21:FooClass.PROPERTY_NAME:"Attribute name ""PROPERTY_NAME"" doesn't conform to snake_case naming style":INFERENCE
 invalid-name:116:4:116:30:FooClass.ABSTRACT_PROPERTY_NAME:"Attribute name ""ABSTRACT_PROPERTY_NAME"" doesn't conform to snake_case naming style":INFERENCE
 invalid-name:121:4:121:28:FooClass.PROPERTY_NAME_SETTER:"Attribute name ""PROPERTY_NAME_SETTER"" doesn't conform to snake_case naming style":INFERENCE

